src/components/DonutPath.js

```diff
@@ -0,0 +1,65 @@
+const React = require('react');
+const d3 = require('d3');
+
+const DonutPath = React.createClass({
+  propTypes: {
+    arc: React.PropTypes.func,
+    color: React.PropTypes.string,
+    opacity: React.PropTypes.number,
+    value: React.PropTypes.object
+  },
+
+  getInitialState () {
+    return {
+      elapsedTime: 0,
+      endAngle: 0
+    };
+  },
+
+  componentDidMount () {
+    if (this.props.animateOnLoad) {
+      this.interval = setInterval(this._renderAnimation, 10);
+    }
+  },
+
+  componentWillUnmount () {
+    if (this.props.animateOnLoad) {
+      clearInterval(this.interval);
+    }
+  },
+
+  _renderAnimation () {
+    let elapsedTime = this.state.elapsedTime;
+    let endAngle = this.state.endAngle;
+
+    endAngle += (elapsedTime / 5000);
```

Comment:
Right now this is for the animation timing, which doesn't work. When that is finished, it will be passed as an animationDuration prop.